### DISCUSSION TITLE
node_modules unnecessarily excluded in tsconfig

### DISCUSSION THREAD

**Discussion by darichey** [vercel/next.js#35879] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

```
    Operating System:
      Platform: linux
      Arch: x64
      Version: #55-Ubuntu SMP Wed Jan 12 17:36:34 UTC 2022
    Binaries:
      Node: 17.8.0
      npm: 8.5.5
      Yarn: 1.22.18
      pnpm: N/A
    Relevant packages:
      next: 12.1.5-canary.0
      react: 18.0.0
      react-dom: 18.0.0
```

### What browser are you using? (if relevant)

Not relevant.

### How are you deploying your application? (if relevant)

Not relevant.

### Describe the Bug

Next unnecessarily writes `"exclude": ["node_modules"]` when creating a new TypeScript project (i.e., with `npx create-next-app@latest --ts`) or to the project's existing `tsconfig.json`. 

```
We detected TypeScript in your project and reconfigured your tsconfig.json file for you. Strict-mode is set to false by default.

The following suggested values were added to your tsconfig.json. These values can be changed to fit your project's needs:

        - exclude was set to ['node_modules']
```

### Expected Behavior

Next should not do this because:
1. TypeScript already ignores `node_modules` by default (see https://www.typescriptlang.org/tsconfig#exclude).
2. This configuration is _incorrect_ if `node_modules` is not located in the current directory (as is the case when using npm workspaces, for example).

### To Reproduce

1. Create a new Next prject with `npx create-next-app@latest --ts`

Or

1. Run `next dev` in an existing Next TypeScript project where `exclude` is not defined in the project's `tsconfig.json`.

**Comment by balazsorban44** [comment]
1. I believe we are doing this to be explicit, but there might actually be a historical reason also. I could not find anything on it, so let me ask the team. :+1:
2. You can override `exclude` for use cases like that.

I don't think it counts as a bug though, so I converted this to a discussion.

**Comment by scamden** [comment]
I'm inheriting a much more robust exclude from an extended tsconfig.json. Could you all at least read the extends before overriding? This cause things not to be excluded that should and I'd rather not duplicate this.

**Reply by ozyman42** [reply]
I don't think you can inherit anything outside of compilerOptions when extending.
